server/routes/notification.js

```diff
@@ -0,0 +1,23 @@
+const express = require('express');
+const router = express.Router();
+const protect = require('../middleware/auth');
+
+// add these controllers when this ticket is finished
+
+// const {
+//   retrieveUsersNotifications,
+//   retrieveUsersUnreadNotifications,
+//    createNotification,
+//    markAsRead
+// } = require('../controllers/notification');
+
+// router.route('/retrieveUsersNotifications').get(protect, retrieveUsersNotifications);
```

Comment:
Nice routes! We can further simplify our routes as the HTTP methods would depict the purpose of a route :)

```
router.route('/all').get(protect, retrieveUsersNotifications);
router.route('/unread').get(protect, retrieveUsersUnreadNotifications);
router.route('/').post(protect, createNotification);
router.route('/mark-read/:notificationId').patch(protect, markAsRead);
```
Some minor comments here:
- We use hyphenated case for route names in JS
- Use the HTTP verb 'PATCH' when we only update part of a resource